src/orion/algo/dehb/dehb.py

```diff
@@ -0,0 +1,509 @@
+
+
+# pylint: disable=too-many-public-methods
+class DEHB(DEHBImpl, BaseAlgorithm):
+    """Differential Evolution with HyperBand
+
+    This class is a wrapper around the librairy DEHB:
+    https://github.com/automl/DEHB.
+
+    For more information on the algorithm,
+    see original paper at https://arxiv.org/abs/2105.09821.
+
+    Awad, Noor, Neeratyoy Mallik, and Frank Hutter. "Dehb: Evolutionary hyperband for scalable,
+    robust and efficient hyperparameter optimization." arXiv preprint arXiv:2105.09821 (2021).
+
+    Parameters
+    ----------
+    space: `orion.algo.space.Space`
+        Optimisation space with priors for each dimension.
+
+    seed: None, int or sequence of int
+        Seed for the random number generator used to sample new trials.
+        Default: ``None``
+
+    mutation_factor: float
+        Mutation probability
+        Default: ``0.5``
+
+    crossover_prob: float
+        Crossover probability
+        Default: ``0.5``
+
+    mutation_strategy: str
+        Mutation strategy rand1, rand2dir randtobest1 currenttobest1 best1 best2 rand2
+        Default: ``'rand1'``
+
+    crossover_strategy: str
+        Crossover strategy bin or exp
+        Default: ``'bin'``
+
+    boundary_fix_type: str
+        Boundary fix method, clip or random
+        Default: ``'random'``
+
+    min_clip: float
+        Min clip when boundary fix method is clip
+        Default: ``None``
+
+    max_clip: float
+        Max clip when boundary fix method is clip
+        Default: ``None``
+
+    max_age: Optional[int]
+        Default: ``np.inf``
+
+    """
+
+    requires_type: ClassVar[Optional[str]] = None
+    requires_dist = None
+    requires_shape = "flattened"
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        space: Space = None,
+        seed: Optional[int] = None,
+        mutation_factor: float = 0.5,
+        crossover_prob: float = 0.5,
+        mutation_strategy: str = "rand1",
+        crossover_strategy: str = "bin",
+        boundary_fix_type: str = "random",
+        min_clip: Optional[int] = None,
+        max_clip: Optional[int] = None,
+        max_age: Optional[int] = np.inf,
+    ):
+        # Sanity Check
+        if mutation_strategy not in MUTATION_STRATEGIES:
+            raise UnsupportedConfiguration(
+                f"Mutation strategy {mutation_strategy} not supported"
+            )
+
+        if crossover_strategy not in CROSSOVER_STRATEGY:
+            raise UnsupportedConfiguration(
+                f"Crossover strategy {crossover_strategy} not supported"
+            )
+
+        if boundary_fix_type not in FIX_MODES:
+            raise UnsupportedConfiguration(
+                f"Boundary fix type {boundary_fix_type} not supported"
+            )
+
+        # We need the transformed algo to initialize DEHB
+        # so store the arguments for after the constructor
+
+        self._original = space
```

Comment:
Not true anymore! This might be misleading, I suggest we adapt this so it doesn't have this kind of two-stage init anymore.

Reply:
Fixed in b5775969